specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,236 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-autorest";
+import "@azure-tools/typespec-azure-core";
+
+enum ServiceApiVersions {
+  v2023_06_01_preview: "2023-06-01-preview"
+}
+
+@useAuth(
+  ApiKeyAuth<ApiKeyLocation.header, "SharedAccessKey"> | OAuth2Auth<[{
+    type: OAuth2FlowType.implicit,
+    authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+    scopes: ["https://eventgrid.azure.net/.default"],
+  }]>
+)
+
+@service({
+  title: "Azure.Messaging.EventGridClient",
+  version: "2023-06-01-preview"
+})
+
+//
+// Supported operations.
+//
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive?api-Version={apiVersion}&timeout=60&maxEvents={maxEvents}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:acknowledge?api-Version={apiVersion}
+// POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:release?api-version={apiVersion}
+//
+
+@server(
+  "{endpoint}",
+  "The host name of the topic",
+  {
+    @doc("The host name of the namespace, e.g. namespaceName1.westus-1.eventgrid.azure.net")
+    endpoint: url,
+  }
+)
+
+@doc("Azure Messaging EventGrid Client")
+@versioned(ServiceApiVersions)
+namespace Azure.Messaging.EventGrid {
+  using TypeSpec.Http;
+  using TypeSpec.Rest;
+  using TypeSpec.Versioning;
+  using Azure.Core;
+  using Azure.Core.Foundations;
+
+  @doc("Properties of an event published to an Azure Messaging EventGrid Namespace topic using the CloudEvent 1.0 Schema.")
+  model CloudEventEvent {
+    @doc("An identifier for the event. The combination of id and source must be unique for each distinct event.")
+    id: string;
+
+    @doc("Identifies the context in which an event happened. The combination of id and source must be unique for each distinct event.")
+    source: string;
+
+    @doc("Event data specific to the event type.")
+    data?: unknown;
+
+    @doc("Event data specific to the event type, encoded as a base64 string.")
+    data_base64?: bytes;
+    
+    @doc("Type of event related to the originating occurrence.")
+    type: string;
+
+    @doc("The time (in UTC) the event was generated, in RFC3339 format.")
+    time?: utcDateTime;
+
+    @doc("The version of the CloudEvents specification which the event uses.")
+    specversion: string;
+
+    @doc("Identifies the schema that data adheres to.")
+    dataschema?: string;
+
+    @doc("Content type of data value.")
+    datacontenttype?: string;
+
+    @doc("This describes the subject of the event in the context of the event producer (identified by source).")
+    subject?: string;
+  }
+
+  @doc("LockToken information.")
+  model LockToken {
```

Comment:
We can't have the model and property name be the same thing. Can this be called something like LockTokenProperties?

Reply:
Also, possibly too late for the initial preview, but I really wonder whether it is worth having this model vs simply defining the lockToken as a string on each of the containing models. Especially considering we don't even use this model for both input and output (only output).